### PR TITLE
Some hints to "Developer Guideline" for beginning.

### PR DESCRIPTION
**Pull request by SandorDobos**
Some paragraphs added to the promising "Developer Guideline".

## Description
During actual development I collect some information on how to do various things.
Publishing it in some form may help others to spare time for actual development.

## Motivation and Context

- Adds some documentation

## Types of changes

- [ ] Improvement (correction, content improvement, typoe fix, formating)
- [x] New Article (new document for docs website)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)

## Checklist:

- [x] My doc follows the code style of this project.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.

### CODE REVIEW

File: content/contributing/developer-guide.md [hunk 1]
```diff
@@ -1,3 +1,148 @@
 # Developer Guideline
```
Comment (networkfusion) [inline content/contributing/developer-guide.md:1]: Should be `Developer Guidelines`

File: content/contributing/developer-guide.md [hunk 2]
```diff
@@ -1,3 +1,148 @@
...
 
 We're working on it! Stay tuned! Raise PR and that will help us finding good recommendations.
+
+Till then here follows some unsorted tips.
```
Comment (networkfusion) [inline content/contributing/developer-guide.md:5]: Until, rather than `Till`

Comment (networkfusion) [reply content/contributing/developer-guide.md:5]: infact: `Until then, here follows some unsorted tips.`

File: content/contributing/developer-guide.md [hunk 3]
```diff
@@ -1,3 +1,148 @@
...
+Neither are a full reference but they give some clue where to look next.
```
Comment (networkfusion) [inline content/contributing/developer-guide.md:6]: `These are not a full reference, but they give some clues on where to look next.`

File: content/contributing/developer-guide.md [hunk 4]
```diff
@@ -1,3 +1,148 @@
...
+   String numberGroupSeparator,
+   int[] numberGroupSizes);
+```
+
+2. Add code which calls the function above as you wish.
+3. Build the solution.
+4. Compare the ```nanoFramework.CoreLibrary\bin\Debug\Stubs``` folder's actual state with the saved one. The files which should have changed:
+- ```corlib_native.cpp```
+- ```corlib_native.h```
+- your class's C++ counterpart, ```corlib_native_System_Number.cpp``` in the my example
```
Comment (networkfusion) [inline content/contributing/developer-guide.md:33]: remove the word `my`

File: content/contributing/developer-guide.md [hunk 5]
```diff
@@ -1,3 +1,148 @@
...
+5. Apply the changes you found to same files under ```nf-interpreter/src/CLR/CorLib```. DO NOT overwrite the files there! The files under nf-interpreter may have additional declarations, etc. Copy over the diff meaningfully!
```
Comment (networkfusion) [inline content/contributing/developer-guide.md:34]: `Apply the changes you found to THE same files under`

File: content/contributing/developer-guide.md [hunk 6]
```diff
@@ -1,3 +1,148 @@
...
+6. You will find that a stub for the function you declared above will be generated with this signature:
+
+```
+HRESULT Library_corlib_native_System_Number::
+    FormatNative___STATIC__STRING__OBJECT__BOOLEAN__STRING__STRING__STRING__STRING__SZARRAY_I4(CLR_RT_StackFrame &stack)
+```
+
+7. Now you can implement your function.
+
+## How to handle in C++ parameter values received from C# call
```
Comment (networkfusion) [inline content/contributing/developer-guide.md:44]: `## How to handle THE C++ parameter values received from A C# call`

File: content/contributing/developer-guide.md [hunk 7]
```diff
@@ -1,3 +1,148 @@
...
+
+Lets see the example method discussed in (#How-to-call-native-code-from-managed-code) tip. The generated (by lib-CoreLibrary solution build, where you declared your needs as an ```extern``` function) C++ stub will have a ```CLR_RT_StackFrame &stack``` parameter.
```
Comment (networkfusion) [inline content/contributing/developer-guide.md:46]: THE tip

File: content/contributing/developer-guide.md [hunk 8]
```diff
@@ -1,3 +1,148 @@
...
+```
+
+## Returning with an Exception
+
+```
+// see other CLR_E_* defined values
+NANOCLR_SET_AND_LEAVE(CLR_E_FAIL);
+```
+
+Not sure on differences, but there is a ```NotImplementedStub``` helper on ```CLR_RT_StackFrame &stack``` parameter may be used too:
```
Comment (networkfusion) [inline content/contributing/developer-guide.md:121]: ON the differences.

**Review by networkfusion — COMMENTED** [review]
Just a few comments to help improve the grammer. Nothing special.

**Review by josesimoes — APPROVED** [review]
Nice contribution. Thanks!

**Pull request merged**